src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExecutorPluginInformation.cs

```diff
@@ -19,4 +19,19 @@ public TestExecutorPluginInformation(Type testExecutorType)
         {
         }
     }
+
+    /// <summary>
+    /// The test executor 2 plugin information.
+    /// </summary>
+    internal class TestExecutorPluginInformation2 : TestExtensionPluginInformation
```

Comment:
>TestExecutorPluginInformation2 [](start = 19, length = 30)

Perhaps we could add a version info to the type above instead?

Reply:
Really good point. Do you mean something like an attribute or rather a field in the `TestExecutorPluginInformation` class ?

Unfortunately due to the way the `PluginCache` is designed (it caches types it finds in assemblies using the `TestPluginInformation` actual type as a key) this won't work without significant changes. What I mean here is if we, say, ask for all `ITestExecutor` types using `TestExecutorPluginInformation` then the `PluginCache` will search for the requested type in all assemblies. However, when we ask for `ITestExecutor2` using the same `TestExecutorPluginInformation` the cache will conclude it already cached types for this `TestPluginInformation` and will return them from the cache which is incorrect. We still need to go ahead and search for it in the assemblies and this behavior can only be employed if we pass another derived type for `TestPluginInformation` other than `TestExecutorPluginInformation`.

We already eyed some changes to the way we discover types in assemblies for the perf work we're doing on TP side, so I think this may be better fitted there. We can redesign the way the `PluginCache` works to enable the behavior you proposed here.